lib/src/commands/test/test.dart

```diff
@@ -56,21 +81,33 @@ class TestCommand extends Command<int> {
 
   @override
   Future<int> run() async {
-    final recursive = _argResults['recursive'] as bool;
     final targetPath = path.normalize(Directory.current.absolute.path);
+    final pubspec = File(path.join(targetPath, 'pubspec.yaml'));
+
+    if (!pubspec.existsSync()) {
```

Comment:
I m curious why moving this up and not using the exception anymore

Reply:
I wanted to catch this error before running the pre gen hook since it would fail if a pubspec didn't exist.